src/Perlang.Tests.Integration/Arrays/IntArrayTests.cs

```diff
@@ -0,0 +1,52 @@
+using FluentAssertions;
+using Xunit;
+using static Perlang.Tests.Integration.EvalHelper;
+
+namespace Perlang.Tests.Integration.Arrays;
+
+public class IntArrayTests
+{
+    [Fact(Skip = "Explicitly typed arrays are not yet supported")]
+    public void explicitly_typed_int_array_can_be_indexed()
+    {
+        string source = """
+                var a: int[] = [1, 2, 3];
+
+                print a[0];
+                print a[1];
+                print a[2];
+                """;
+
+        var output = EvalReturningOutput(source);
+
+        output.Should()
+            .BeEquivalentTo(
+                "1",
+                "2",
+                "3"
+            );
+    }
+
+    [Fact]
+    public void implicitly_typed_int_array_can_be_indexed()
+    {
+        string source = """
+                var a = [1, 2, 3];
+
+                print a[0];
+                print a[1];
+                print a[2];
+                """;
+
+        var output = EvalReturningOutput(source);
+
+        output.Should()
+            .BeEquivalentTo(
+                "1",
+                "2",
+                "3"
+            );
+    }
+
+    // TODO: Add test for indexing before and after array size, ensuring that we get the expected exceptions
```

Comment:
Hmm, would have been nice to take in the same PR I guess. :see_no_evil: Will have to come in a followup PR.

Reply:
https://github.com/perlang-org/perlang/pull/496/files#r1644983171